crates/sui-core/src/authority.rs

```diff
@@ -965,6 +1039,48 @@ impl AuthorityState {
         }
 
         state
+            .process_recovery_log(None)
+            .await
+            .expect("Could not fully process recovery log at startup!");
+
+        state
+    }
+
+    // Continually pop in-progress txes from the WAL and try to drive them to completion.
+    async fn process_recovery_log(&self, limit: Option<usize>) -> SuiResult {
+        let mut limit = limit.unwrap_or(usize::max_value());
+        while limit > 0 {
+            limit -= 1;
+            if let Some(tx_guard) = self.database.wal.pop_one_recoverable_tx().await {
```

Comment:
Yes, it is not deleted until you call `tx_guard.commit_tx()` (because you might fail while trying to recover it, so we would need to make another attempt).

Any suggestions for other names? I have to use a one-at-a-time interface here, because there's no guarantee that you can hold the locks for more than one TX from the recovery log.